pandas/tests/scalar/timedelta/test_timedelta.py

```diff
@@ -550,7 +550,7 @@ def test_overflow(self):
 
         # mean
         result = (s - s.min()).mean()
-        expected = pd.Timedelta((pd.DatetimeIndex((s - s.min())).asi8 / len(s)
+        expected = pd.Timedelta((pd.TimedeltaIndex((s - s.min())).asi8 / len(s)
```

Comment:
Why was this incorrect?  How was the mistake caught in the first place?

Reply:
Caught it when making it warn in #23675.